python/dgl/nn/pytorch/conv/edgegraphconv.py

```diff
@@ -0,0 +1,293 @@
+            must contain two tensors of shape :math:`(N_{in}, D_{in_{src}})` and
+            :math:`(N_{out}, D_{in_{dst}})`.
+        weight : torch.Tensor, optional
+            Optional external weight tensor.
+        edge_weight : torch.Tensor, optional
+            Optional tensor on the edge. If given, the convolution will weight
+            with regard to the message. This should of shape :math:`(E, P)` where E is the number 
+            of edged and P is the dimension of the edge weight.
+
+        Returns
+        -------
+        torch.Tensor
+            The output feature
+
+        Raises
+        ------
+        DGLError
+            Case 1:
+            If there are 0-in-degree nodes in the input graph, it will raise DGLError
+            since no message will be passed to those nodes. This will cause invalid output.
+            The error can be ignored by setting ``allow_zero_in_degree`` parameter to ``True``.
+
+            Case 2:
+            External weight is provided while at the same time the module
+            has defined its own weight parameter.
+
+        Note
+        ----
+        * Input shape: :math:`(N, *, \text{in_feats})` where * means any number of additional
+          dimensions, :math:`N` is the number of nodes.
+        * Output shape: :math:`(N, *, \text{out_feats}) * \text{edge_weight_dimension}` where all but the last dimension are
+          the same shape as the input.
+        * Weight shape: :math:`(\text{in_feats}, \text{out_feats})`.
+        * Edge weight shape: :math:`(\text{num_edges}, \text{edge_weight_dimension})`.
+        """
+        with graph.local_scope():
+            if not self._allow_zero_in_degree:
+                if (graph.in_degrees() == 0).any():
+                    raise DGLError('There are 0-in-degree nodes in the graph, '
+                                   'output for those nodes will be invalid. '
+                                   'This is harmful for some applications, '
+                                   'causing silent performance regression. '
+                                   'Adding self-loop on the input graph by '
+                                   'calling `g = dgl.add_self_loop(g)` will resolve '
+                                   'the issue. Setting ``allow_zero_in_degree`` '
+                                   'to be `True` when constructing this module will '
+                                   'suppress the check and let the code run.')
+            if edge_weight is None:
+                raise DGLError("Edge weight is required for this layer type (EdgeGraphConv)")
+            assert edge_weight.shape[0] == graph.number_of_edges()
+            aggregate_fn = fn.u_mul_e('h', '_edge_weight', 'm')
+
+            # (BarclayII) For RGCN on heterogeneous graphs we need to support GCN on bipartite.
+            feat_src, feat_dst = expand_as_pair(feat, graph)
+            if self._norm in ['left', 'both']:
```

Comment:
@abigail-katcoff-kensho  do you understand why they normalize on the left before the following operations and on the right after?